meshtastic/interfaces/ble/lifecycle_receive_runtime.py

```
@@ -0,0 +1,459 @@
+
+class BLEReceiveLifecycleCoordinator:
+    """Own receive-loop intent and receive-thread lifecycle behavior."""
+
+    def __init__(self, iface: "BLEInterface") -> None:
+        """Bind receive lifecycle ownership to a specific interface."""
+        self._iface = iface
+        self._thread_access = _LifecycleThreadAccess(iface)
+        self._deferred_restart_lock = threading.Lock()
+        self._deferred_restart_inflight = False
+
+    def _schedule_deferred_receive_restart(
```

Ruff (0.15.6)

[warning] Too many statements (56 > 50)

(PLR0915)